gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -121,22 +124,37 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
     projectSelectorViewModel = getViewModel(ProjectSelectorViewModel.class);
   }
 
+  private void onFeatureAdded(Feature feature) {
+    feature.getLayer().getForm().ifPresent(formId -> addNewObservation(feature));
+  }
+
+  private void addNewObservation(Feature feature) {
+    String projectId = feature.getProject().getId();
+    String featureId = feature.getId();
+    String formId = feature.getLayer().getForm().get().getId();
+    navigator.addObservation(projectId, featureId, formId);
+  }
+
+  /** This is only possible after updating the location of the feature. So, reset the UI. */
   private void onFeatureUpdated(Boolean result) {
     if (result) {
-      Toast.makeText(getContext(), "Done", Toast.LENGTH_SHORT).show();
       mapContainerFragment.setDefaultMode();
-    } else {
-      Timber.e("Failed to update feature");
     }
   }
 
+  /** Hide the bottom sheet as deleting a feature also removes child observations. */
```

Comment:
This Javadoc isn't needed since the code is self-explanatory, and the method is internal/private.